src/qt/glcanvas.cpp

```diff
@@ -411,6 +443,9 @@ bool wxGLCanvas::Create(wxWindow *parent,
 
     m_qtWindow = new wxQtGLWidget(parent, this, format);
 
+    QGestureRecognizer* pPanRecognizer = new PanGestureRecognizer();
+    QGestureRecognizer::registerRecognizer(pPanRecognizer);
+
```

Comment:
I don't know if there is any drawback in doing this unconditionally or if we should only do when touch events are enabled.

Maybe a comment here could help clarify things?

Reply:
If a custom gesture recognizer is required for a particular type of wxWindow, we need to create and register it.  Theoretically, we could implement EnableTouchEvents(int eventsMask) in this class, which is a wxWindow subclass, and defer creation of the recognizer to that point.  However, Qt provides no interface to determine if a particular recognizer has already been created and registered.  So we might have to resort to some static variable to retain the existence state of the custom recognizer.  Not really a good idea...
A comment in the code could say:
//  Create and register a custom recognizer, available to all instances of this class.

I cannot see any drawback to always creating the recognizer.  If it is never enabled (by EnableTouchEvents()), then it will be dormant and automatically deleted on app exit.